core/src/main/java/org/apache/druid/java/util/metrics/AllocationMetricCollector.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.metrics;
+
+import org.apache.druid.java.util.common.logger.Logger;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+class AllocationMetricCollector
+{
+  private static final Logger log = new Logger(AllocationMetricCollector.class);
+  private Method getThreadAllocatedBytes;
+  private ThreadMXBean threadMXBean;
+  private Map<Long, Long> previousResults;
+  private boolean initialized = false;
+
+  AllocationMetricCollector()
+  {
+    try {
```

Comment:
Since this is a delta metric, maybe it makes more sense to start with an empty map. So the first call to `calculateDelta()` accounts all data emitted since the Druid application start.

Reply:
✅